packages/core/src/ui/InlineToolbar/index.ts

```diff
@@ -123,14 +132,28 @@ export class InlineToolbar {
 
     this.#toolbar = make('div');
 
-    Array.from(this.#tools.keys()).forEach((toolName) => {
+    this.#tools.forEach((tool, toolName) => {
       const inlineElementButton = make('button');
 
       inlineElementButton.innerHTML = toolName;
 
-      inlineElementButton.addEventListener('click', (_event) => {
-        this.apply(toolName as InlineToolName);
-      });
+      console.log(tool, tool.hasActions);
+
+      /**
+       * If tool has actions, then on click of the element button we should render actions element
+       * If tool has no action, then on click of the element button we should apply format
+       */
+      if (tool.hasActions) {
```

Comment:
you can move this statement inside event listener

Reply:
this is check for UI part (would be separated later) i think, that there is no need to move this in event listener (business logic part)